webapp/platform/design-system/src/components/primitives/tag/tag.tsx

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React, {forwardRef, memo, useMemo} from 'react';
+import type {ReactNode, MouseEventHandler} from 'react';
+import {useIntl} from 'react-intl';
+
+import WithTooltip from '../with_tooltip';
+
+import './tag.scss';
+
+export type TagSize = 'xs' | 'sm' | 'md' | 'lg';
+
+export type TagVariant =
+    | 'default'
+    | 'info'
+    | 'success'
+    | 'warning'
+    | 'danger'
+    | 'dangerDim'
+    | 'primary'
+    | 'secondary';
+
+export type TagPreset = 'beta' | 'bot' | 'guest' | 'custom';
+
+export interface TagProps {
+
+    /** The text content of the tag */
+    text?: ReactNode;
+
+    /** Predefined tag type (beta, bot, guest, or custom for manual text) */
+    preset?: TagPreset;
+
+    /** Size variant of the tag */
+    size?: TagSize;
+
+    /** Visual variant/color scheme of the tag */
+    variant?: TagVariant;
+
+    /** Whether to display text in uppercase */
+    uppercase?: boolean;
+
+    /** Icon element to display before the text (React component) */
+    icon?: ReactNode;
+
+    /** Icon size in pixels - will be auto-calculated based on tag size if not provided */
+    iconSize?: number;
```

Comment:
Why make this configurable? I think it would be better to minimize the number of additional props that these components need